pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -839,6 +825,36 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 						Service:                    service,
 					}
 
+					// Hack for Kubernetes stateful sets/headless services
+					// Instead of generating a single 0.0.0.0:Port listener, generate N listeners
+					// (where N <= 5 at most) one for each podIP.
+					if bind == "" && service.Resolution == model.Passthrough &&
+						service.Attributes.ServiceRegistry == string(serviceregistry.KubernetesRegistry) {
+						instances, err := env.InstancesByPort(service, servicePort.Port, nil)
+						if err != nil {
+							// we can't do anything. Fallback to the usual 0.0.0.0:Port listener
+							goto standardListenerLogic
+						}
+						if len(instances) > 5 || len(instances) == 0 {
```

Comment:
nit: make this a comment at the top with comments.

Reply:
`5` seems kind of arbitrary. I don't really want to add yet another obscure option to control this, but won't we run into situations where users have 5 instances, all is good, then they scale up to 6 and everything blows up?